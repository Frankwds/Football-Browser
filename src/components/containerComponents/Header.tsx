import React from "react";
import { Heading } from "@chakra-ui/react"
import football from "./../../resources/football.jpg";

const acc = "rgb(0,128,128)";
/**
 * React class component displaying the header of the application
 * 
 * @returns {React.Component}
 */
class Header extends React.Component {
  render() {
    return (
      <div id={"header"}>
        <Heading bg={acc} color={"white"} textAlign="center">WebDev Project3 - Football browser</Heading>
        <img className={"imgR"} src={football} alt="football icon" />
        
      </div>
    );
  }
}

export default Header;
