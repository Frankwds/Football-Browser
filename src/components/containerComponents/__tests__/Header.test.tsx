import ReactDOM from "react-dom";
import Header from "../Header";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";


let container: any;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("Testing Header", () => {
  it("should render without crashing", () => {
    act(() => {
      const div = document.createElement("div");
      ReactDOM.render(<Header />, div);
    })
  });

  it("snapshot should be same as previous", () => {
    const tree = renderer
      .create(<Header />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

