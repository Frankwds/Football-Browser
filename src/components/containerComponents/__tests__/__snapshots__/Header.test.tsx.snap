// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header snapshot should be same as previous 1`] = `
<div
  id="header"
>
  <h2
    className="chakra-heading css-176yoig"
  >
    WebDev Project3 - Football browser
  </h2>
  <img
    alt="football icon"
    className="imgR"
    src="football.jpg"
  />
</div>
`;
